plugins/ns/timeleft.sma

```diff
@@ -62,8 +66,8 @@ public sayTheTime(id){
   }
   new ctime[64]  
   get_time("%m/%d/%Y - %H:%M:%S",ctime,charsmax(ctime))  
-  client_print(0,print_chat, "The time:   %s",ctime )
-  return PLUGIN_CONTINUE
+  client_print(get_pcvar_bool(g_amx_time_print_to_all) ? 0 : id,print_chat, "The time:   %s",ctime )
+  return get_pcvar_bool(g_amx_time_show_cmd_in_chat) ? PLUGIN_CONTINUE : PLUGIN_HANDLED
```

Comment:
Could be just !get_pcvar_bool(g_amx_time_show_cmd_in_chat).

Reply:
@OciXCrom it is ok, it should be explicit.

Reply:
I didn't say it's not ok("could be" not "should be").  But the line is too long and the shorter version is clear enough, seeing get_pcvar_BOOL clearly tells it will either return 0 or 1.

Reply:
But you don't want to reutrn 0 or 1, you want `PLUGIN_CONTINUE` or `PLUGIN_HANDLED` (yes, I know they are the same underneath).

Reply:
It's fine as it is. I agree `PLUGIN_` should be explicit. Just because it happens to be 0 and 1 doesn't mean you should abuse this behavior. If the long line is an issue, you can always cache the car's value before, totally fine.

Reply:
Of course there are more solutions, like caching as you said/even breaking the line. Also, I would not call "abusing" taking advantage of behavior that is set in stone.

Reply:
`PLUGIN_` are symbols with their definitions and the documentation says you have to use them. If you don't explicit them, you imply their associated values that you are not supposed to use directly because irrelevant, not useful and confusing. If a beginner sees a code which returns some value from a condition, he may not understand why, whereas if you use `PLUGIN_*`, It's visually easier to understand and it's documented.
That's why if you do that, you're abusing with shortcuts, which is unhelpful.

Reply:
You are talking about explicitly returning numerical values instead of PLUGIN_* which I never said should be done. What I am saying is that I believe it is fine to return directly the result of an expression/native when doing that can save you from complicated/verbose/long lines of code.

Reply:
@HamletEagle but there is a magic in converting between `bool` and `PLUGIN_*`.

Reply:
@WPMGPRoSToTeMa, the approach is not perfect, but lines that are too verbose are not perfect either. This could be solved in multiple other ways, as stated before: caching/breaking the line/using if-else instead of ternary,

Reply:
You explicit the value underneath by returning a boolean. Those constants could have held others values, like -1, or 2. Actually, what about ``PLUGIN_HANDLED_MAIN`, are you going to return some boolean + 1? This just abusing the behavior for the fake sake of simplicity and you make harder to understand what's happening. As stated above, don't use magic.

Simplifying doesn't mean shorter and condensed code.

About the code, you should not use `LANG_PLAYER` with an index.

Also to make the code easier to understand, you can:
* Bind the cvars to variables
* Remove few terniary operator

Possible example:

```SourcePawn
public sayTimeLeft(id)
{
    new printTarget = id;

    if (CvarPrintToAll)
    {
        printTarget = 0;
        SetGlobalTransTarget(LANG_PLAYER);
    }

    if (CvarTimeLimit > 0.0)
    {
        if (CvarTimeVoice)
        {
            new voice[128];
            setTimeVoice(voice, charsmax(voice), 0, get_timeleft());
            client_cmd(id, "%s", voice);
        }

        client_print(printTarget, print_chat, "%l:  %d:%02d", "TIME_LEFT", (a / 60), (a % 60)):
    }
    else
    {
        client_print(printTarget, print_chat, "%l", "NO_T_LIMIT"):
    }

    return CvarShowCommandInChat ? PLUGIN_CONTINUE : PLUGIN_HANDLED:
}
```

Reply:
I meant "modernization" - using the new functions where possible.